python/ray/dashboard/utils.py

```diff
@@ -334,12 +325,14 @@ def get_all_modules(module_type):
                 )
                 raise e
 
+    if is_in_test() and not should_only_load_minimal_modules:
+        import ray.dashboard.modules.tests.test_head  # noqa: F401
+        import ray.dashboard.modules.tests.test_agent  # noqa: F401
```

Comment:
This means all `ray.init()` in the tests will still be slow?

Reply:
We should only load them for tests that need them.

Reply:
Okay then I'll add back the `RAY_DASHBOARD_MODULE_TEST` environment variable.

Reply:
But I don't think it'll slow to much, because we only import `test_head` and `test_agent`, unlike the `__init__.py`, which will import all files under the `ray/dashboard/tests` folder.

Reply:
ok, let's test it out. If it's not slow, we can keep as it is.

Reply:
Compared to the `premerge` of #52289, it seems that the total execution time of `core: python tests` are roughly the same.

![image](https://github.com/user-attachments/assets/34ebd075-d4ea-4c48-b285-af47378c765f)
![image](https://github.com/user-attachments/assets/9e1c7566-1d9c-4f47-8b23-1a70deaae0d2)

Reply:
But that might not be precise. Let me know if I need to open another PR to test this.

Reply:
You can do the same test as done in #52289

Reply:
- master branch, 3.89 sec ![image](https://github.com/user-attachments/assets/0a33f9f5-1b39-42ec-9d3b-550f55fd9e0a)
- This PR, 3.38 sec 
![image](https://github.com/user-attachments/assets/c936a3cc-4865-4748-93ff-b9461038e2de)
- This PR, loading `test_head` and `test_agent` by removing `is_in_test()` check in `get_all_modules`, 3.38 sec 
![image](https://github.com/user-attachments/assets/f1e09685-1ee6-4bc7-bf5f-48c63e52e4b9)

So it seems there's no slowdown.